graphql_compiler/schema.py

```diff
@@ -275,3 +277,61 @@ def _parse_datetime_value(value):
     RecurseDirective,
     FoldDirective,
 )
+
+
+TYPENAME_META_FIELD_NAME = '__typename'  # This meta field is built-in.
+COUNT_META_FIELD_NAME = '__count'
+
+
+ALL_SUPPORTED_META_FIELDS = frozenset((
+    TYPENAME_META_FIELD_NAME,
+    COUNT_META_FIELD_NAME,
+))
+
+
+EXTENDED_META_FIELD_DEFINITIONS = OrderedDict((
+    (COUNT_META_FIELD_NAME, GraphQLField(GraphQLInt)),
+))
+
+
+def is_meta_field(field_name):
+    """Return True if the field is considered a meta field in the schema, and False otherwise."""
+
+
+def insert_meta_fields_into_existing_schema(graphql_schema):
```

Comment:
Without this being called, the `__count` property if queried will result in a `String does not validate: [GraphQLError('Cannot query field "__count" on type "Foo".',)]` error. There seems to be a little bit of a disconnect between that error and the fact that `__count` is a meta-field that needs to be injected into the schema, and 
So the two things I'm wondering about here are: 
1. What are your thoughts of injecting by default into the schema (maybe through a compiler supplied `build_compiler_schema_ast` to drop-in replace the graphql core `build_schema_ast`?
2. Should the error message around missing meta fields specifically be improved to indicate that this might be the root cause of the issue, ie `This error could be caused because the schema was not augmented with insert_meta_fields_into_existing_schema` or something along those lines to help with the solution?

Reply:
This is a great catch with great suggestions.

- I'm a bit wary of "magically" injecting into the schema by default, without the user being aware that their schema is being modified. I think it's reasonable to make the user use either the provided `insert_meta_fields_into_existing_schema()` function, or the `EXTENDED_META_FIELD_DEFINITIONS` constant when building the schema, since both of those amount to one "edit" to the traditional workflow of building the schema, the same as replacing `build_schema_ast()` with a custom `build_compiler_schema_ast()`.

- I'm definitely open to trying to make the error message around missing meta fields steer the user in the right direction. I'm going to add a bit more documentation around inserting the meta fields into the schema. Would you mind if I open a `user friendliness` issue for catching the "missing meta field" error and reraising it with some more useful info, since that might not be trivial to do and might take some work? I'd prefer not to block this PR because of that, since it's not immediately clear to me how much work doing that would be, and because that's a "you didn't follow the documentation, here's what you missed" sort of error improvement that I think is important but not a blocker.